contracts/native-staking/src/contract.rs

```
@@ -49,8 +60,24 @@ impl NativeStakingContract<'_> {
     }
 
     #[msg(query)]
-    fn config(&self, _ctx: QueryCtx) -> Result<ConfigResponse, ContractError> {
-        todo!()
+    fn config(&self, ctx: QueryCtx) -> Result<ConfigResponse, ContractError> {
+        self.config.load(ctx.deps.storage).map_err(Into::into)
+    }
+
+    fn reply_init_callback(
+        &self,
+        deps: DepsMut,
+        reply: SubMsgResponse,
+    ) -> Result<Response, ContractError> {
+        let init_data = parse_instantiate_response_data(&reply.data.unwrap())?;
+
+        // Associate staking proxy with owner address
+        let proxy_addr = Addr::unchecked(init_data.contract_address);
+        let owner_data: OwnerMsg = from_slice(&init_data.data.unwrap())?; // TODO?: Check for None
```

I would avoid unwrapping. Better to raise some missing data or invalid data errors. On the other hand, it is the response to instantiation, so we control the contract ourselves, so neither of those should fire (especially the first one), still - it is called once, and checking (at least the second one) really won't hurt gas-wise, and will report a bug better.

---

Same thing. I prefer `data.ok_or(ContractError::NoInstatiateData)?`